fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java

```diff
@@ -513,9 +516,23 @@ private SumoVehicleState processVehicleSubscriptionResult(final long time,
                     // for parking vehicles, there are no consumptions and emissions to measure
                     .consumptions(new VehicleConsumptions(
                             new Consumptions(0d), sumoVehicle.lastVehicleData.getVehicleConsumptions().getAllConsumptions())
-                    ).emissions(new VehicleEmissions(
-                            new Emissions(0d, 0d, 0d, 0d, 0d), sumoVehicle.lastVehicleData.getVehicleEmissions().getAllEmissions())
-                    );
+                    ).emissions(
+                            new VehicleEmissions(new Emissions(0d, 0d, 0d, 0d, 0d),
+                                    sumoVehicle.lastVehicleData.getVehicleEmissions().getAllEmissions()));
+            if (isWaitingToLeaveParking) {
+                log.debug("Vehicle {} is currently waiting to leave parking area on edge {}.",
+                        veh.id, sumoVehicle.lastVehicleData.getRoadPosition().getConnectionId());
+                // if the vehicle is waiting to leave a parking area, we assume it's still parked and copy previous vehicle data
+                vehicleDataBuilder
+                        .position(sumoVehicle.lastVehicleData.getPosition(), sumoVehicle.lastVehicleData.getProjectedPosition())
+                        .movement(0.0d, 0.0d, fixDistanceDriven(-1, sumoVehicle.lastVehicleData))
+                        .orientation(DriveDirection.UNAVAILABLE, sumoVehicle.lastVehicleData.getHeading(), sumoVehicle.lastVehicleData.getSlope())
+                        .route(sumoVehicle.lastVehicleData.getRouteId())
+                        .signals(sumoVehicle.lastVehicleData.getVehicleSignals())
+                        .stopped(sumoVehicle.lastVehicleData.getVehicleStopMode())
+                        .sensors(sumoVehicle.lastVehicleData.getVehicleSensors())
+                        .laneArea(sumoVehicle.lastVehicleData.getLaneAreaId());
+            }
```

Comment:
That's true, what I tried to do was avoiding setting `consumptions, emissions, ...` twice, as it is handled the same way as vehicles being parked.
I'm not sure if we could just use `.copyFrom(...)` for the usual parking case as well, this would make the code quite a bit prettier as we could handle `isParking` and `isWaitingToLeaveParkingArea` the same way.

Reply:
Go ahead, try and we'll see :)